apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -245,6 +245,8 @@ def fabric_l3(self):
         if self._fabric_l3 is None:
             self._fabric_l3 = self._cisco_l3 or hasattr(self._l3_plugin,
                                                         '_apic_driver')
+            # Update the apicapi's property
+            self.apic_manager.apic_config.enable_aci_routing = self._fabric_l3
```

Comment:
I definitely was worried about that. It ends up that it's okay, but there's a different issue - this isn't the state that needs updating. I've just pushed an amended pull-request that modifies this, as well as it's dependent variable in apicapi. 

The reason I went this route is because I was worried about the lack of ordering between the L3 plugin and the ML2 plugin. The _fabric_l3 flag is lazily loaded on first request, which makes it more likely for the L3 plugin to be available so that we can verify that one of our L3 Plugins (APIC ML2 or GBP) is in use.

If the concern about this is from a best practice point of view (i.e. I can see an argument for considering this to be immutable state after passing it to __init__), then I can explore alternatives.